File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjCSharp/Model/TicketPurchase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Model
{
    [Serializable]
    public class TicketPurchase : Entity<long>
    {
        public Game Game { get; set; }
        public int NoOfTickets { get; set; }
        public float TotalPrice { get; set; }
        public string ClientName { get; set; }

        public TicketPurchase() : base(0)
        {
            Game = null;
            NoOfTickets = 0;
            TotalPrice = 0;
            ClientName = "";
        }

        public TicketPurchase(Game game, int noOfTickets, float totalPrice, string clientName): base(0)
        {
            Game = game;
            NoOfTickets = noOfTickets;
            TotalPrice = totalPrice;
            ClientName = clientName;
        }

        public override string ToString()
        {
            return "TicketPurchase: " + base.Id + " for game " + Game.Id + ", " +
                NoOfTickets + " bought by " + ClientName + " for " + TotalPrice + "$";
        }

        public override bool Equals(object obj)
        {
            return obj is TicketPurchase purchase &&
                   Id == purchase.Id &&
                   EqualityComparer<Game>.Default.Equals(Game, purchase.Game) &&
                   NoOfTickets == purchase.NoOfTickets &&
                   TotalPrice == purchase.TotalPrice &&
                   ClientName == purchase.ClientName;
        }

        public override int GetHashCode()
        {
            int hashCode = 202997279;
            hashCode = hashCode * -1521134295 + Id.GetHashCode();
            hashCode = hashCode * -1521134295 + EqualityComparer<Game>.Default.GetHashCode(Game);
            hashCode = hashCode * -1521134295 + NoOfTickets.GetHashCode();
            hashCode = hashCode * -1521134295 + TotalPrice.GetHashCode();
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(ClientName);
            return hashCode;
        }
    }
}

File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjJavaFX/Model/src/main/java/app/model/TicketPurchase.java
package app.model;

import java.io.Serializable;
import java.util.Objects;

public class TicketPurchase extends Entity<Long> implements Serializable {
    private Game game;
    private Integer noOfTickets;
    private Float totalPrice;
    private String clientName;

    public TicketPurchase() {
        this.game = null;
        this.noOfTickets = 0;
        this.totalPrice = 0F;
        this.clientName = "";
    }

    public TicketPurchase(Game game, Integer noOfTickets, Float totalPrice, String clientName) {
        this.game = game;
        this.noOfTickets = noOfTickets;
        this.totalPrice = totalPrice;
        this.clientName = clientName;
    }

    public Game getGame() {
        return game;
    }

    public void setGame(Game game) {
        this.game = game;
    }

    public Integer getNoOfTickets() {
        return noOfTickets;
    }

    public void setNoOfTickets(Integer noOfTickets) {
        this.noOfTickets = noOfTickets;
    }

    public Float getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(Float totalPrice) {
        this.totalPrice = totalPrice;
    }

    public String getClientName() {
        return clientName;
    }

    public void setClientName(String clientName) {
        this.clientName = clientName;
    }

    @Override
    public String toString() {
        return "TicketPurchase: " + super.getId() + " , game " + game.getId() + ", " +
                noOfTickets + " tickets bought by " + clientName + " for " + totalPrice + "$";
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TicketPurchase that = (TicketPurchase) o;
        return Objects.equals(game, that.game) && Objects.equals(noOfTickets, that.noOfTickets) &&
                Objects.equals(totalPrice, that.totalPrice) && Objects.equals(clientName, that.clientName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(game, noOfTickets, totalPrice, clientName);
    }
}

